eurostag-converter/src/main/java/com/powsybl/eurostag/converter/EurostagEchExport.java

```diff
@@ -562,9 +566,315 @@ private void createTransformers(EsgNetwork esgNetwork, EsgGeneralParameters para
+            RatioTapChanger rtcL2 = t3wt.getLeg2().getRatioTapChanger();
+            PhaseTapChanger ptcL2 = t3wt.getLeg2().getPhaseTapChanger();
+
+            EsgThreeWindingTransformer.RegulatingMode regulatingMode = EsgThreeWindingTransformer.RegulatingMode.NOT_REGULATING;
+            Esg8charName zbusr = null; //...regulated node name (if empty, no tap change)
+            double voltr = Double.NaN;
+            int ktpnom = 1; //...nominal tap number is not available in IIDM. Take th median plot by default
+            int ktap8 = 1;  //...initial tap position (tap number) (Ex: 10)
+            List<EsgThreeWindingTransformer.Tap> taps = new ArrayList<>();
+
+            int rtci = 0;
+            int nbTaps = 1; //there is at least one tap
+            if (rtcL1 != null && rtcL1.isRegulating()) {
+                rtci = 1;
+                nbTaps = rtcL1.getStepCount();
+                ConnectionBus regulatingBus = ConnectionBus.fromTerminal(rtcL1.getRegulationTerminal(), config, null);
+                if (regulatingBus.getId() != null) {
+                    regulatingMode = EsgThreeWindingTransformer.RegulatingMode.VOLTAGE;
+                    zbusr = new Esg8charName(dictionary.getEsgId(regulatingBus.getId()));
+                    voltr = rtcL1.getTargetV();
+                    ktap8 = rtcL1.getTapPosition() - rtcL1.getLowTapPosition() + 1;
+                    ktpnom = rtcL1.getStepCount() / 2 + 1;
+                }
+            } else if (rtcL2 != null && rtcL2.isRegulating()) {
+                rtci = 2;
+                nbTaps = rtcL2.getStepCount();
+                ConnectionBus regulatingBus = ConnectionBus.fromTerminal(rtcL2.getRegulationTerminal(), config, null);
+                if (regulatingBus.getId() != null) {
+                    regulatingMode = EsgThreeWindingTransformer.RegulatingMode.VOLTAGE;
+                    zbusr = new Esg8charName(dictionary.getEsgId(regulatingBus.getId()));
+                    voltr = rtcL2.getTargetV();
+                    ktap8 = rtcL2.getTapPosition() - rtcL2.getLowTapPosition() + 1;
+                    ktpnom = rtcL2.getStepCount() / 2 + 1;
+                }
+            } else if (rtcL3 != null && rtcL3.isRegulating()) {
+                rtci = 3;
+                nbTaps = rtcL3.getStepCount();
+                ConnectionBus regulatingBus = ConnectionBus.fromTerminal(rtcL3.getRegulationTerminal(), config, null);
+                if (regulatingBus.getId() != null) {
+                    regulatingMode = EsgThreeWindingTransformer.RegulatingMode.VOLTAGE;
+                    zbusr = new Esg8charName(dictionary.getEsgId(regulatingBus.getId()));
+                    voltr = rtcL3.getTargetV();
+                    ktap8 = rtcL3.getTapPosition() - rtcL3.getLowTapPosition() + 1;
+                    ktpnom = rtcL3.getStepCount() / 2 + 1;
+                }
+            }
+
+            double[] ucc12 = new double[nbTaps];
```

Comment:
@JB-H Why do you need tables here and not simple double variables to ease the understanding of your code?

Reply:
It is because I don't know which ones will be constant and those with a different value per tap (1, 2 or 3?)
In theory I only need one table for the leg with a ratio tap changer and a constant value for the others, so you are right

Reply:
you are right, tables are not necessary...